### PR TITLE
Stats: Release the new 7-day highlights feature

### PR DESCRIPTION
**Pull request by jsnmoon**
#### Proposed Changes

image

* Releases the new 7-day highlights section for the Traffic Stats page.
* Adds trend arrows to each highlight card.

#### Testing Instructions

* Spin up this change in a live branch.
* Navigate to Stats.
* Ensure you see the new 7-day highlights section.

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #69224.

### CODE REVIEW

**Review by dognose24 — APPROVED** [review]
LGTM! However, I wonder if the [percentage's divisor](https://github.com/Automattic/wp-calypso/blob/c3ce23d5678733008ff4d609e26e3e8e5d1392f8/packages/components/src/highlight-cards/highlight-card.tsx#L46) should be the `previousCount`.

**Comment by jsnmoon** [comment]
> I wonder if the [percentage's divisor](https://github.com/Automattic/wp-calypso/blob/c3ce23d5678733008ff4d609e26e3e8e5d1392f8/packages/components/src/highlight-cards/highlight-card.tsx#L46) should be the `previousCount`.

I wager it's more intuitive to contextualize the difference as a percentage of the current count since the current count is front and center in every highlight card.

You could argue that using the previous count for percentage calculation will contextualize how values change historically. For example, if the views go from 100 to 10,100, that'll be an increase of either `99%` (current count percentage) or `10,000%` (previous count percentage).

However, doing this will inevitably spawn "Why does it say 10,000% instead of 99%?" customer questions since it's not *obvious* how the percentage calculation is done.

**Comment by jeffgolenski** [comment]
@jsnmoon You brilliant human being, you. This is looking wonderful! The functionality is spot on—I've just got a few style tweaks for you. Here's an image with the various scenarios, a list of changes, and as always you can see the details on the      PR Reviews / adjustments Figma page.

PR 69862

- [ ] Text: “Compared to last seven days” should become an info icon (gray 50) with a tooltip. Tooltip should include a more robust description and visual. See proposed design. (let me know if that svg visual won't work in the tooltip component)
- [x] Icons: for “views” should be the eye svg we just imported for tooltips and a few other spots!
- [x] Text: “Visitors” “Views” “Likes” “Comments” should be the san-serif calypso is using, not Recoleta. (apple system or Inter, I believe)
- [x] Counts: The actual number counts “60” “75” “0” “0” should be Recoleta typeface
- [ ] Counts: When we hit the larger numbers with numerical abbreviation (K, M), we’ll want to build a tooltip in to show full amount on hover. See proposed designs
- [x] Trend icons: Should be up or down arrows (part of the wordpress icon set!)
- [x] Trends functionality: If a trend is stagnant, like 0 comments past 14 days and 0 comments past week, don’t display anything (see comments card above in the proposed design). If count is 0 for the past 7 days, but there’s a downward trend over past 14 days, we’ll still want to display the down arrow (see likes card above). 
- [x] Other: There’s a random gray strip at the bottom of the current implementation. Looks like it’s caused by overflow-x: scroll. We don’t want the scroll bar to appear at all—even on mobile!

Let me know if any clarification is needed around any of that stuff, my friend! cc @keoshi

**Comment by jsnmoon** [comment]
Thanks for the careful review, @jeffgolenski! I've addressed most of the issues you've brought up and have two questions:

1) Due to how we've set up the new highlight cards component, importing the tooltip component is currently nontrivial. Do you consider the missing header tooltip and the truncated count tooltip release blockers?
2) Do you know when/where we added the eye SVG?

**Review by keoshi — COMMENTED** [review]
This is incredible, thank you so much @jsnmoon !

> Compared to last seven days

In the meantime while we don't update this to be an icon + tooltip, I believe we should be more precise. It's not actually “compared to the last seven days” but compared to the 7 days before that. 

- [x] That's why we suggested "Compared to previous 7 days", indicating we're taking a look at a two week period and comparing how the evolution looks like.


> Trends functionality: If a trend is stagnant, like 0 comments past 14 days and 0 comments past week, don’t display anything (see comments card above in the proposed design). If count is 0 for the past 7 days, but there’s a downward trend over past 14 days, we’ll still want to display the down arrow (see likes card above).

Don't know if I agree with this @jeffgolenski. Hear me out: because this module is about trends, highlighting there's been no change is actually valuable. Plus, it makes the module more consistent, always showing relevant trends regardless if they're positive, negative, or null. In that sense it would make sense to show `0 (0%)` and my preference would be to show no icon at all. What do you think?

> Mobile

Noting a couple of things here:

- [x] The trends numbers, absolute and relative, are being hidden in smaller resolutions.
- [x] The module is touching the browser's limits (i.e.: no spacing). We should probably do two things here: add left/right margin for the title and add padding in between the cards to create that negative space.

PR | Mock
--|--
image | image

> Do you consider the missing header tooltip and the truncated count tooltip release blockers?

IMO not at all. I see we have a `title` attribute in there which will allow users to see the exact metrics so that will cover most scenarios.

image

As for the header tooltip, I consider that an extra and we can we always clarify later.


> Do you know when/where we added the eye SVG?

The eye icon is not yet part of the WP library. 

- [x] While we work on adding it there, could we use the SVG code itself? Here it is:

```
<svg width="24" height="24" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="m4 13 .67.336.003-.005a2.42 2.42 0 0 1 .094-.17c.071-.122.18-.302.329-.52.298-.435.749-1.017 1.359-1.598C7.673 9.883 9.498 8.75 12 8.75s4.326 1.132 5.545 2.293c.61.581 1.061 1.163 1.36 1.599a8.29 8.29 0 0 1 .422.689l.002.005L20 13l.67-.336v-.003l-.003-.005-.008-.015-.028-.052a9.752 9.752 0 0 0-.489-.794 11.6 11.6 0 0 0-1.562-1.838C17.174 8.617 14.998 7.25 12 7.25S6.827 8.618 5.42 9.957c-.702.669-1.22 1.337-1.563 1.839a9.77 9.77 0 0 0-.516.845l-.008.015-.002.005-.001.002v.001L4 13Zm8 3a3.5 3.5 0 1 0 0-7 3.5 3.5 0 0 0 0 7Z" fill="#00101C"/></svg>
```

Finally, a couple more things on icons:

- [x] Comments: can we use the `comment-content` icon instead? It should add a few lines inside the comment box, making it look more like a comment and less like a chat bubble.
- [x] Trends: could we decrease the size of the arrows a bit so the icon matches the copy? Here's a comparison:

PR | Mock
--|--
image | image

Here's the changes I've applied:

```diff
.highlight-card-difference {
+  align-items: center;
}

<Icon
+  size={ 18 }
/>
```

**Comment by jeffgolenski** [comment]
@jsnmoon Brilliant. thanks for jumping on those so quickly. To answer your Q's

> Do you consider the missing header tooltip and the truncated count tooltip release blockers?

They are not blockers. Full speed ahead.

> Do you know when/where we added the eye SVG?

Filipe just supplied the svg in his comment above. But I was referencing our slack convo from a few weeks ago. (I believe you just commented in that thread a couple of hours before this comment).

**Comment by jeffgolenski** [comment]
> Don't know if I agree with this @jeffgolenski. Hear me out: because this module is about trends, highlighting there's been no change is actually valuable. Plus, it makes the module more consistent, always showing relevant trends regardless if they're positive, negative, or null. In that sense it would make sense to show 0 (0%) and my preference would be to show no icon at all. What do you think?

@keoshi Now that you've explained it, I agree with you. @jsnmoon Ignore my comment above about hiding the trends when they're 0 -> 0. Filipe made a good argument.

**Review by kangzj — APPROVED** [review]
looking sharp

Web E2E failed but seems non-relevant. Restarted.

**Comment by jeffgolenski** [comment]
@jsnmoon Well done my friend. Just a few minor styling bits of feedback. Since this is a new feature we're releasing, I want to ensure it's at 💯 

**Large and medium screens**
PR 69863 part 2 - large

Text: “Compared to the last seven days”

- [x] Change typeface to the same as “Visitors, Views, Like, Comments” sub heads
- [x] Change color to Gray 60
- [x] Font weight 400

- [x] Align trend arrows and numbers (%) flush with the bottom of number counts. See red line above.
- [x] Icon: Change bubble icon above “Comments” to the “comment-content” icon in the WordPress Icon Library (almost the same, but the new one has lines in it).

Medium screens: 

- [x] The trend numbers overlap the bounding box of the card. Min width should be the total width of all the elements in the card + 16px.  (Adding `display: inline-block` to .highlight-card-difference seems to fix this in my quick tests)
- [x] Behavior gets weird on tablet sized screens

**Mobile specific**
PR 69863 part 2 - mobile

- [x] Alignment: Ensure container has 24px left and right padding to align with title and sub title.
- [x] Trends: Numbers and % aren’t showing on small screens. Let’s show everything by the icons here, puh-lease.

(all styling, such as typography, color, etc should be changed on all screens, please)

I'll let @keoshi chime in on anything I may have missed, before we ship!

**Comment by jsnmoon** [comment]
@jeffgolenski: Those screenshots appear to be from an older checkout of this branch; the comments icon should have already been updated! Could you double check? I went ahead and updated styles where still applicable.

EDIT: You may want to ensure that the "Docker image" check has passed before clicking on the live branch link 🙂

**Comment by keoshi** [comment]
Loving the changes and how this is looking. Absolutely beautiful from the loading state to the mobile views.

Final couple of notes from my side:

#### Rounded numbers

Noticed that we're rounding up some numbers. In the example below you'll see that `2K` is presented but the real value is `1989`. In this example I'd expect to see 1.9K.

- [x] What's the rounding math on these?

image

#### Cards being cut off

The entire section looks great on mobile! The only exception is that the cards are not flush with the window, so they appear cut off.

- [x] Could we move the margin from `.highlight-cards` to `.highlight-cards-heading` and
- [x] Use the same value of `1em` for padding around the cards in `.highlight-cards-list`?

image

Thanks for being patient with us!

**Comment by jeffgolenski** [comment]
@jsnmoon 

> Those screenshots appear to be from an older checkout of this branch; the comments icon should have already been updated! Could you double check? I went ahead and updated styles where still applicable. You may want to ensure that the "Docker image" check has passed before clicking on the live branch link 🙂

Awesome tip! Sorry about that. I'll keep it in mind moving forward. 

Just gave it a go again. Aside from the feedback Filipe provided above, this LGTM. Thanks for all your work here. Freakin' awesome! 

Screen Shot 2022-11-11 at 12 08 40 PM

**Comment by jsnmoon** [comment]
>  What's the rounding math on these?

We're using the standard `Intl.NumberFormat` in [short, compact notation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#scientific_engineering_or_compact_notations). The [default behavior](https://stackoverflow.com/questions/66043475/default-rounding-strategy-in-intl-numberformat-prototype-format) for the formatter is to use

**Comment by jsnmoon** [comment]
>  What's the rounding math on these?

We're using the standard `Intl.NumberFormat` in [short, compact notation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#scientific_engineering_or_compact_notations). The default behavior for the formatter is to use:
- No fractional digits (123.**456**)
- Minimum of 1 integer digit (12**3**.456)
- Minimum of 1 & maximum of 2 significant digits (**1**23.456)

For the example you mentioned, I believe `1989` is being rounded to `2.0K`, but the trailing fractional zero is being dropped for brevity. I'd ideally like to see `2.0K` here, but it's something we can punt for later.

> [Other review feedback]

Every other piece of feedback has been addressed!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7621234

Thank you @jsnmoon for including a screenshot in the description! This is really helpful for our translators.

File: client/my-sites/stats/all-time-highlights-section/style.scss
Comment (jsnmoon) [inline client/my-sites/stats/all-time-highlights-section/style.scss:4]: These styles were bleeding onto the 7-day highlights (and probably onto annual highlights), so I limited the scope of these styles to the all time highlights section. Let's try our best to avoid regressions like this in the future (cc @dognose24).

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7621234

Thank you @jsnmoon for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7621234).

**Pull request merged**
